packages/core/src/api/exporters/markdown/util/convertVideoToMarkdownRehypePlugin.ts

```diff
@@ -0,0 +1,18 @@
+import { visit } from "unist-util-visit";
+
+// Originally, rehypeParse parses videos as links, which is incorrect.
+export function convertVideoToMarkdown() {
+  return (tree: any) => {
+    visit(tree, "element", (node, index, parent) => {
+      if (node.tagName === "video") {
+        const src = node.properties?.src || node.properties?.["data-url"] || "";
+        const name =
+          node.properties?.title || node.properties?.["data-name"] || "";
+        parent.children[index!] = {
+          type: "text",
+          value: `![${name}](${src})`,
+        };
```

Comment:
Based on the test cases, I don't know that this is working entirely correctly.

It is being rendered to a link here: tests/src/unit/core/formatConversion/export/__snapshots__/html/video.html which I don't fully get why. So, it makes sense why this parse would not actually pick it up.

This may have to do with the way we output internal html vs external HTML?

I think we should hold off on merging this until we change up the default block API in https://github.com/TypeCellOS/BlockNote/pull/1904